core/embed/sec/tropic/tropic.c

```diff
@@ -74,49 +119,19 @@ bool tropic_init(void) {
   // length was chosen arbitrarily. A shorter delay may be sufficient.
   hal_delay(100);
 
-#ifdef TREZOR_EMULATOR
-  pkey_index_t pairing_key_slot = TROPIC_FACTORY_PAIRING_KEY_SLOT;
-#else
-  pkey_index_t pairing_key_slot = TROPIC_PRIVILEGED_PAIRING_KEY_SLOT;
-#endif
-
-  curve25519_key trezor_privkey = {0};
-  secbool privkey_ok = secret_key_tropic_pairing_privileged(trezor_privkey);
-  if (privkey_ok != sectrue) {
-    pairing_key_slot = TROPIC_UNPRIVILEGED_PAIRING_KEY_SLOT;
-    privkey_ok = secret_key_tropic_pairing_unprivileged(trezor_privkey);
-  }
-
-  curve25519_key tropic_pubkey = {0};
-  secbool pubkey_ok = secret_key_tropic_public(tropic_pubkey);
-
-#if !PRODUCTION
-  // Allow running with default factory keys in non-production fw
-  if (privkey_ok != sectrue) {
-    tropic_get_factory_privkey(trezor_privkey);
-    pairing_key_slot = TROPIC_FACTORY_PAIRING_KEY_SLOT;
-    privkey_ok = sectrue;
+#ifndef TREZOR_EMULATOR
+  if (session_start(drv, TROPIC_UNPRIVILEGED_PAIRING_KEY_SLOT)) {
```

Comment:
We should start with the privileged key and then try the unprivileged.

Reply:
Fixed in https://github.com/trezor/trezor-firmware/pull/5845/commits/4bfb81f38d23d17515b67559d7114f8ae2c9d51f.